deploy/services/service_setup.py

```diff
@@ -0,0 +1,168 @@
+                        target_priv.unlink()
+                    latest_priv.rename(target_priv)
+
+                if full_candidates:
+                    latest_full = max(full_candidates, key=lambda p: p.name)
+                    target_full = certs_target / "fullchain.pem"
+                    if target_full.exists():
+                        target_full.unlink()
+                    latest_full.rename(target_full)
+
+                return True, f"Certificates copied and normalized in {certs_target}"
+
+            except Exception as e:
+                return False, f"Error copying/normalizing certificates: {e}"
+
+        elif self.os_type == "windows":
+            return False, "Windows: Not implemented."
+        else:
+            return False, f"Unsupported OS: {self.os_type}" 
+
+        
+    def permissions__mongoDB(self) -> Tuple[bool, str]:
+        """
+        Combine privkey.pem + fullchain.pem -> combined.pem and set permissions for MongoDB.
+        """
+
+        if self.os_type in ("linux", "darwin"):
+            try:
+                is_root = (os.geteuid() == 0)
+            except AttributeError:
+                is_root = False
+            if not is_root:
+                return False, ("This operation requires root privileges, run with sudo.")
+
+        try:
+            with open(self.combined, "wb") as out_f: 
+                # cat privkey.pem fullchain.pem > combined.pem
+                with open(self.privkey, "rb") as pk:
+                    out_f.write(pk.read())
+                with open(self.fullchain, "rb") as fc:
+                    out_f.write(fc.read())
+
+            # Set permission mode 600 (owner read/write only)
+            self.combined.chmod(0o600)
+            # Set ownership to mongodb's UID:GID inside container (999:999)
+            if self.os_type in ("linux", "darwin"):
+                subprocess.run(["chown", "999:999", str(self.combined)], check=True)
+            return True, f"combined.pem created at {self.combined} with mode 600 and ownership set."
+        except Exception as e:
+            return False, f"Error creating combined.pem: {e}"
+
+
+
+    def permissions__rabbitMQ(self) -> Tuple[bool, str]:
+        """
+        Copy privkey.pem -> privkey-rabbitmq.pem and set owner to 999 (user-only).
+        """
+        if not self.privkey.exists():
+            return False, f"Missing {self.privkey}; run obtain_tls_certs first."
+
+        if self.os_type in ("linux", "darwin"):
+            try:
+                is_root = (os.geteuid() == 0)
+            except AttributeError:
+                is_root = False
+            if not is_root:
+                return False, ("This operation requires root privileges, run with sudo.")
+        try:
+            # Copy to a new file.
+            shutil.copy2(self.privkey, self.rabbit_key)
+            if self.os_type in ("linux", "darwin"):
+                subprocess.run(["chown", "999", str(self.rabbit_key)], check=True)
```

Comment:
these permissions can be read from `config/services.env` file. It is better to read as much as possible from this config file.